packages/web3-eth-abi/src/types.ts

```diff
@@ -0,0 +1,57 @@
+export interface JsonAbiParameter {
+	readonly name: string;
+	readonly type: string;
+	readonly baseType?: string;
+	readonly indexed?: boolean;
+	readonly components?: Array<JsonAbiParameter>;
+	readonly arrayLength?: number;
+	readonly arrayChildren?: Array<JsonAbiParameter>;
+}
+
+export interface JsonAbiStruct {
+	[key: string]: unknown;
+	name?: string;
+	type: string;
+}
+
+export interface JsonAbiCoderStruct extends JsonAbiStruct {
+	[key: string]: unknown;
+	components?: Array<JsonAbiStruct>;
+}
+
+type FragmentTypes = 'constructor' | 'event' | 'function';
+
+export interface JsonAbiBaseFragment {
+	[key: string]: unknown;
```

Comment:
why we need `[key: string]: unknown;` for any of `'constructor' | 'event' | 'function'` ?

Reply:
It was needed to pass any arbitrary JOSN object to any function. It's already been refactored in next PR so can remove it.